clippy_lints/src/doc/mod.rs

```diff
@@ -378,6 +403,11 @@ impl<'tcx> LateLintPass<'tcx> for Documentation {
         let Some(headers) = check_attrs(cx, &self.valid_idents, attrs) else {
             return;
         };
+
+        if headers.empty && !item.span.is_dummy() {
+            empty_docs::check(cx, attrs);
+        }
```

Comment:
This should go in `check_attrs` so that it's also checked for docs on the crate root, trait items and impl items. Also means we don't need a new field on `DocHeaders`

Reply:
@Alexendoo check_attrs doesn't have access to item, i need to check that the item span is not dummy. Should I add a parameter? also can you please check the tests and tell me if something like
```
Enum Alphabet {
///
A,
B
}
```
should be listed? if so I would have to return the whole item span and need the item for that

Reply:
When would the item span be dummy?

Reply:
I got two cases in my tests, can report back later to what they are.

Reply:
@Alexendoo not sure what other item.kind can be dummy like this and not visible in the file.
```rust
        if empty && item.span.is_dummy() {
            dbg!(item);
        }
```
```rust
[clippy_lints/src/doc/mod.rs:412:13] item = Item {
    ident: #0,
    owner_id: DefId(0:1 ~ empty_docs[0eca]::{use#0}),
    kind: Use(
        Path {
            span: no-location (#1),
            res: [
                Err,
            ],
            segments: [
                PathSegment {
                    ident: std#1,
                    hir_id: HirId(DefId(0:1 ~ empty_docs[0eca]::{use#0}).1),
                    res: Def(
                        Mod,
                        DefId(1:0 ~ std[60a2]),
                    ),
                    args: None,
                    infer_args: false,
                },
                PathSegment {
                    ident: prelude#1,
                    hir_id: HirId(DefId(0:1 ~ empty_docs[0eca]::{use#0}).2),
                    res: Def(
                        Mod,
                        DefId(1:47 ~ std[60a2]::prelude),
                    ),
                    args: None,
                    infer_args: false,
                },
                PathSegment {
                    ident: rust_2021#1,
                    hir_id: HirId(DefId(0:1 ~ empty_docs[0eca]::{use#0}).3),
                    res: Def(
                        Mod,
                        DefId(1:134 ~ std[60a2]::prelude::rust_2021),
                    ),
                    args: None,
                    infer_args: false,
                },
            ],
        },
        Glob,
    ),
    span: no-location (#1),
    vis_span: no-location (#1),
}
[clippy_lints/src/doc/mod.rs:412:13] item = Item {
    ident: std#1,
    owner_id: DefId(0:2 ~ empty_docs[0eca]::std),
    kind: ExternCrate(
        None,
    ),
    span: no-location (#1),
    vis_span: no-location (#1),
}```

Reply:
Those are fine as they don't have doc comments on them

re: the enum you wouldn't need the item span since you only need to point at the doc comments

Reply:
for some reason that doesn't appear in the attrs, I made change you suggest but now all the attrs that don't point at the main item like in the enum above won't get linted

Reply:
this in `Documentation` fixes the issues, not sure how many other check_* there are to do 
```rust
    fn check_variant(&mut self, cx: &LateContext<'tcx>, variant: &'tcx hir::Variant<'tcx>) {
        let attrs = cx.tcx.hir().attrs(variant.hir_id);
        check_attrs(cx, &self.valid_idents, attrs);
    }

    fn check_field_def(&mut self, cx: &LateContext<'tcx>, variant: &'tcx hir::FieldDef<'tcx>) {
        let attrs = cx.tcx.hir().attrs(variant.hir_id);
        check_attrs(cx, &self.valid_idents, attrs);
    }
```
should implement like this?

Reply:
Oh I see! Yeah great catch, as the dogfood shows that means other lints weren't firing on fields/variants

I believe they are the only two that were missing